src/vast-client-serializer.js

```diff
@@ -0,0 +1,703 @@
+        '#': universalAdId.value,
+      }
+    }
+  }
+
+  /**
+   * Build CreativeExtension nodes
+   * @param {Array[Object]} creativeExtensions An array of creativeExtensions objects
+   * @returns {Object} An array of CreativeExtension nodes
+   */
+  buildCreativeExtensions(creativeExtensions) {
+    if (!creativeExtensions) {
+      return null
+    }
+
+    return {
+      'CreativeExtension': creativeExtensions.map(
+        (creativeExtension) => this.buildExtension(creativeExtension)
+      ),
+    }
+  }
+
+  /**
+   * Build Creative node from it's type
+   * @param {Object} creative A creative object
+   * @returns {Object} Either a Linear, NonLinearAds, CompanionAds node
+   */
+  buildCreativeByType(creative) {
+    switch (creative.type) {
+      case 'linear':
+        return {
+          'Linear': this.buildLinear(creative)
+        }
+      case 'nonlinear':
+        return {
+          'NonLinearAds': this.buildNonLinearAds(creative)
+        }
+      case 'companion':
+        return {
+          '@required': creative.required,
+          'CompanionAds': this.buildCompanionAds(creative)
+        }
+    }
+  }
+
+  /**
+   * Build Linear node from creative
+   * @param {Object} creative A creative object
+   * @returns {Object} A Linear node
+   */
+  buildLinear(creative) {
+    return {
+      '@skipoffset': this.convertToHHMMSS(creative.skipDelay),
+      'Duration': this.convertToHHMMSS(creative.duration),
+      'AdParameters': this.buildAdParameters(creative.adParameters),
+      'MediaFiles': this.buildMediafiles(
+        creative.mediaFiles,
+        creative.mezzanine,
+        creative.interactiveCreativeFile,
+        creative.closedCaptionFiles
+      ),
+      'VideoClicks': this.buildVideoClicks(
+        creative.videoClickThroughURLTemplate,
+        creative.videoClickTrackingURLTemplates,
+        creative.videoCustomClickURLTemplates,
+      ),
+      'TrackingEvents': this.buildTrackingEvents(creative.trackingEvents),
+      'Icons': this.buildIcons(creative.icons),
+    }
+  }
+
+  /**
+   * Build AdParameters node
+   * @param {String} adParameters The adParameters string
+   * @param {Boolean} xmlEncoded The xmlEncoded parameter
+   * @returns {Object} A AdParameters node
+   */
+  buildAdParameters(adParameters, xmlEncoded = null) {
+    if (!adParameters) {
+      return null
+    }
+
+    return {
+      '@xmlEncoded': xmlEncoded,
+      '#': this.wrapCDATA(adParameters),
+    }
+  }
+
+  /**
+   * Build VideoClicks parameter node
+   * @param {Object} clickThrough The clickThrough object
+   * @param {Array[Object]} clickTrackings The array of clickTracking objects
+   * @param {Array[Object]} customClicks The array of customClicks objects
+   * @returns {Object} A VideoClicks node
+   */
+  buildVideoClicks(clickThrough, clickTrackings, customClicks) {
+    let videoClicks = {}
+
+    if (clickThrough) {
+      const clickThroughObj = clickThrough === Object ? clickThrough : {
```

Comment:
Shouldn't we use `typeof clickThrough === 'object' `?

Reply:
Nice catch I was probably tired when I wrote this 😅